Keypad.c

```diff
@@ -0,0 +1,223 @@
+/*
+ * Keypad.c
+ *
+ *  Created on: Feb 2, 2020
+ *      Author: Delaney Berger
+ */
+
+#include "Keypad.h"
+#include "delay.h"
+#include "msp.h"
+#include <stdint.h>
+#include "UART.h"
+
+static uint8_t key;
+
+void KeypadInit(void)
+{
+   P5 -> DIR &= ~(C1|C2|C3|C4);                 //Set Columns(input) as interrupts
+   P5 -> IES |= (C1|C2|C3|C4);
+   P5 -> IFG &= ~(C1|C2|C3|C4);
+   P5 -> IE |= (C1|C2|C3|C4);
+   P5 -> SEL1 &= ~(C1|C2|C3|C4);
+   P5 -> SEL0 &= ~(C1|C2|C3|C4);
+   P5 -> REN = (C1|C2|C3|C4);                   // use enable resistor
+   P5 -> OUT &= ~(C1|C2|C3|C4);                 //  pull down resistors
+
+
+   P2 -> DIR = (R0|R1|R2|R3);                   //sets rows as outputs
+/*
+   P2 -> SEL1 &= ~(R0|R1|R2|R3);                //Not sure why but it doesn't work as well with this section of code
+   P2 -> SEL0 &= ~(R0|R1|R2|R3);
+   P2 -> OUT |= (R0|R1|R2|R3);                  // sets it high
+*/
+   __enable_irq();
+   NVIC -> ISER[1] = 1 << ((PORT5_IRQn) & 31);
+}
+void PORT5_IRQHandler(void)                     //interrupt handler on Port 5
+{
+   P5->IFG &= ~(C1|C2|C3|C4);                   //clear interrupt flag
+   keypad_setkey();
+}
+
+void keypad_setkey(void)                        //function that checks the rows and columns for which button was pressed
+{
+    P2 -> OUT &= ~(R0|R1|R2|R3);                //set all low
+    uint8_t COL = 0;
+
+    P2 -> OUT = R0;
+    delay_ms(25, CLK);
+    COL = P5 -> IN & (C1|C2|C3|C4);
+    if (COL != 0)                               //Multiple if statements that goes through the rows to select the correct key based on column
+    {
+        if ((COL & C1) != 0)
+        {
+            key = 1;
+        }
+        else if ((COL & C2) !=0)
+        {
+            key = 2;
+        }
+        else if ((COL & C3) != 0)
+        {
+            key = 3;
+        }
+        else
+        {
+            key = 65;
```

Comment:
I would change all the no hit values (65,66,67,46) to a single value to represent that there was no hit on the keypad

Reply:
Nevermind, looked up an ascii table, this makes sense. I would leave comments at these numbers to make it clear why they were chosen.